tests/cpp/test_multidevice_matmul.cpp

```diff
@@ -109,6 +101,11 @@ TEST_F(DistributedMatmulTest, LayoutTN_NoComms) {
   }
   b->setDeviceMesh(mesh);
 
+  // TODO: If c's allocation domain isn't set, it will fail validation at
+  // csrc/device_lower/validation.cpp:419, Vectorized dim for consumer has to be
+  // from a contiguous inner most position.
+  c->setAllocationDomain(c->getLeafDomain(), true);
```

Comment:
I had to set the allocation domain for this TV so that an error about contiguity wasn't raised. I thought the allocation inferring pass would handle this case, since this test has no communication just sharded computation.

Reply:
This is interesting.... Likely the allocation order inference is getting some strange mapping on output.

I opened #2276 and will investigate it.